source/SH3/camera/camera.cpp

```diff
@@ -0,0 +1,131 @@
+/** @file
+ *
+ *  Implementation of camera.hpp
+ *
+ *  @copyright 2017 Palm Studios
+ *
+ *  @date 1-9-2017
+ *
+ *  @author Jesse Buhagiar
+ */
+#include <cmath>
+#include "SH3/camera/camera.hpp"
+#include "glm/gtc/matrix_transform.hpp"
+#include <algorithm>
+#include <boost/algorithm/clamp.hpp>
+
+static constexpr glm::vec3 worldUp = glm::vec3(0, 1, 0); /**< Which axis is considered 'world up'. */
+
+sh3::camera::Camera::Camera(const glm::vec3& pos, angle<float> fov, float aspect, float zNear, float zFar)
+: camPos(pos), camFov(fov), aRatio(aspect), camNear(zNear), camFar(zFar),
+  camFront(glm::vec3(0.0f, 0.0f, -1.0f)), camUp(glm::vec3(0.0f, 1.0f, 0.0f)),
+  pMatrix(glm::perspective(0.0f, aspect, zNear, zFar)),
```

Comment:
We don't need to initialize `pMatrix` and `vMatrix` since that is done in `RecalculatePerspective`.